build_deps.sh

```diff
@@ -25,39 +25,52 @@ llvm_branch=tags/RELEASE_700/final
 klee_repo=https://github.com/rsas/klee
 klee_branch=pure-bv-qf-llvm-7.0
 alive_repo=https://github.com/manasij7479/alive2.git
+z3_repo=https://github.com/Z3Prover/z3.git
+z3_branch=z3-4.8.4
 
 llvm_build_type=Release
 if [ -n "$1" ] ; then
   llvm_build_type="$1"
   shift
 fi
 
+z3_srcdir=$(pwd)/third_party/z3
+z3_installdir=$(pwd)/third_party/z3-install
+git clone $z3_repo $z3_srcdir
+mkdir -p $z3_installdir
+
+(cd $z3_srcdir && git checkout $z3_branch &&
+python scripts/mk_make.py --noomp --prefix=$z3_installdir &&
+cd build && make -j8 install)
+
+export PATH=$z3_installdir/bin:$PATH
+
 alivedir=third_party/alive2
 alive_builddir=$alivedir/build
 mkdir -p $alivedir $alive_builddir
 git clone $alive_repo $alivedir/alive2
 
 if [ -n "`which ninja`" ] ; then
-  (cd $alive_builddir && cmake ../alive2 -DCMAKE_BUILD_TYPE=$llvm_build_type -GNinja)
+  (cd $alive_builddir && cmake ../alive2 -DZ3_LIBRARIES=$z3_installdir/lib/libz3.so -DZ3_INCLUDE_DIR=$z3_installdir/include -DCMAKE_BUILD_TYPE=$llvm_build_type -GNinja)
   ninja -C $alive_builddir
 else
-  (cd $alive_builddir && cmake ../alive2 -DCMAKE_BUILD_TYPE=$llvm_build_type)
-  make -C $alive_builddir -j4
+  (cd $alive_builddir && cmake ../alive2 -DZ3_LIBRARIES=$z3_installdir/lib/libz3.so -DZ3_INCLUDE_DIR=$z3_installdir/include -DCMAKE_BUILD_TYPE=$llvm_build_type)
+  make -C $alive_builddir -j8
```

Comment:
Hm, why is this using make instead of ninja, and thus makes a guess at the appropriate thread count?
This looks like a bug.